File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    basket = []
    answer = 0
    for move in moves:
        doll = 0
        for row in board:
            if row[move-1] != 0:
                doll = row[move-1]
                row[move-1] = 0
                break
        if doll == 0:
            break
            
        if len(basket) == 0:
            basket.append(doll)
        else:
            if basket[-1] == doll:
                basket.pop()
                answer += 2
            else:
                basket.append(doll)
                
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var answer = 0;
    const basket = [];
    const n = board.length
    const boardArr = new Array(n + 1).fill(0).map(_ => []);
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            if (board[i][j] !== 0) boardArr[j + 1].push(board[i][j])
        }
    }
    
    moves.forEach(move => {
        if (basket.length === 0) {
            basket.push(boardArr[move].shift())
        } else {
            const newDoll = boardArr[move].shift()
            const prevDoll = basket.at(-1)
            if (newDoll) {
                if (newDoll === prevDoll) {
                basket.pop()
                answer += 2
                } else {
                basket.push(newDoll)
            }
            }
            
        }

    })
    return answer;
}